Composer/packages/client/src/recoilModel/undo/__test__/undoHistory.test.ts

```diff
@@ -47,12 +50,12 @@ describe('undoHistory class', () => {
     expect(undoHistory.stack.length).toBe(0);
   });
 
-  it('should only support 30 history', () => {
+  it('should only support 40 actions in history', () => {
```

Comment:
This test still looks like it's only testing for 30 actions (it generates 40, but then checks that 10 through 39 are present, so it's throwing out 0 through 9).

Reply:
Nice catch. So we push a max of 30 items on the stack. As we add beyond that we do a stack.shift() to keep pushing the oldest elements out of the stack. I will revert the test name